### ISSUE TITLE
[Bug]: beam_PostCommit_Go_VR_spark is failing

### ISSUE THREAD

**Issue by jrmccluskey** [apache/beam#22933]
### What happened?

The Go Spark PostCommit is perma-red due to an issue in the TestLPErrorPipeline test, outputting this error:

```
2022/08/29 02:50:25  (): java.lang.RuntimeException: Error received from SDK harness for instruction 1: process bundle failed for instruction 1 using plan 204-2 : while executing Process for Plan[204-2]:
2: DataSink[S[fn/write/n8_flatten:1:0@localhost:37955]] Coder:W;fn/wire/n8_flatten:1:0<KV;c11<int[varintz;c2];c3,KV;c10<int[varintz;c2];c3,bytes;c0>>>!GWC
3: PCollection[n8_flatten:1] Out:[2]
4: Flatten[1]. Out:DataSink[S[fn/write/n8_flatten:1:0@localhost:37955]] Coder:W;fn/wire/n8_flatten:1:0<KV;c11<int[varintz;c2];c3,KV;c10<int[varintz;c2];c3,bytes;c0>>>!GWC
5: PCollection[n7_e8_inject1] Out:[4]
6: Inject[1]. Out:5
7: PCollection[n7] Out:[6]
8: ParDo[regression.toFoo] Out:[7]
1: DataSource[S[fn/read/n6:0@localhost:37955], local_output] Coder:W;fn/wire/n6:0<CoGBK;c7<int[varintz;c2];c3,R;c4[*regression.fruit]>>!GWC Out:8
	caused by:
source decode failed
	caused by:
EOF
```

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-go

**Comment by damccorm** [comment]
The first failing build is from this commit - https://github.com/apache/beam/commit/e9089dd99630d939f0c38fbacbe97a283e429fc2 - https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/6517/consoleFull

The last success is from this commit - https://github.com/apache/beam/commit/8347b9e1d36cb8c2a1d863909d2d27a00a3efdaa - https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/6516/consoleFull

So its probably an issue in between those 2 (on august 26 here - https://github.com/apache/beam/commits/e9089dd99630d939f0c38fbacbe97a283e429fc2)

Given that its a coding issue on a datasource, I'm most suspicious of https://github.com/apache/beam/commit/172bd31d01bcc1e87a5aa1e37379a669c81b46bf

**Comment by jrmccluskey** [comment]
That seems like a reasonable assessment

**Comment by lostluck** [comment]
Marking as closed as #23042 seems to have resolved the issue. Knowing this test catches it is good though.

Remember, it's always the right move to rollback/revert if a change causes postcommit test failures.
